qiita_pet/handlers/api_proxy/artifact.py

```diff
@@ -5,8 +5,78 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+from qiita_core.util import execute_as_transaction
+from qiita_pet.handlers.api_proxy.util import check_access, check_fp
 from qiita_db.artifact import Artifact
-from qiita_pet.handlers.api_proxy.util import check_access
+from qiita_db.metadata_template.prep_template import PrepTemplate
+from qiita_db.util import convert_to_id
+
+
+@execute_as_transaction
+def artifact_post_req(user_id, filepaths, artifact_type, name,
+                      prep_template_id):
+    """Creates the prep template and initial artifact for the prep template
```

Comment:
There will be the option to upload the prep template without the initial artifact? There may be cases where this is useful (i.e. I'm adding the prep template while the sequencing is being done, so I can reduce the time to process once the sequences are available)